src/app/task-details/task-details.component.ts

```diff
@@ -13,11 +15,12 @@ import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angula
   styleUrl: './task-details.component.scss'
 })
 export class TaskDetailsComponent implements OnInit {
-  task!: Task;
-  taskForm : FormGroup ;
-  isEditMode= false;
+  task?: Task;
```

Comment:
task shouldn't be optional, since you can't access the page if a task can't be found